src/resolve-ref.ts

```diff
@@ -65,13 +65,16 @@ export async function createRefIndex(gitdir: string) {
 
     const listRemotes = async () => {
         const remotes = [];
-        const entries = await fsPromises.readdir(pathJoin(gitdir, 'refs', 'remotes'), {
-            withFileTypes: true
-        });
+        const remotesDir = pathJoin(gitdir, 'refs', 'remotes');
+        if (existsSync(remotesDir)) {
+            const entries = await fsPromises.readdir(pathJoin(gitdir, 'refs', 'remotes'), {
+                withFileTypes: true
+            });
```

Comment:
Maybe in case of `!existsSync(remotesDir)` just use an empty array?

```suggestion
        const entries = existsSync(remotesDir)
            ? await fsPromises.readdir(pathJoin(gitdir, 'refs', 'remotes'), {
                   withFileTypes: true
               })
            : [];
```